From d9113fd59bec0fdcd03922406a7fcddfe589e72a Mon Sep 17 00:00:00 2001
From: LordPsyan <samccstudios@gmail.com>
Date: Fri, 20 Jan 2012 09:46:22 -0600
Subject: [PATCH] 2012-01-20_Enable-EmeraldForest

---
 src/game/Player.cpp |   21 ++++++++++++---------
 1 files changed, 12 insertions(+), 9 deletions(-)

diff --git a/src/game/Player.cpp b/src/game/Player.cpp
index 8af6f39..bfd8cf4 100644
--- a/src/game/Player.cpp
+++ b/src/game/Player.cpp
@@ -1708,21 +1708,24 @@ bool Player::TeleportTo(uint32 mapid, float x, float y, float z, float orientati
 
     if (GetMapId() != mapid)
     {
-        if (!(options & TELE_TO_CHECKED))
+        if (mapid != 169)
         {
-            if (!CheckTransferPossibility(mapid))
+            if (!(options & TELE_TO_CHECKED))
             {
-                if (GetTransport())
-                    TeleportToHomebind();
+                if (!CheckTransferPossibility(mapid))
+                {
+                    if (GetTransport())
+                        TeleportToHomebind();
 
-                DEBUG_LOG("Player::TeleportTo %s is NOT teleported to map %u (requirements check failed)", GetName(), mapid);
+                    DEBUG_LOG("Player::TeleportTo %s is NOT teleported to map %u (requirements check failed)", GetName(), mapid);
 
-                return false;                                       // normal client can't teleport to this map...
+                    return false;                                       // normal client can't teleport to this map...
+                }
+                else
+                    options |= TELE_TO_CHECKED;
             }
-            else
-                options |= TELE_TO_CHECKED;
+            DEBUG_LOG("Player::TeleportTo %s is being far teleported to map %u", GetName(), mapid);
         }
-        DEBUG_LOG("Player::TeleportTo %s is being far teleported to map %u", GetName(), mapid);
     }
     else
     {
-- 
1.7.2.5

