From 3bc903ffdd86f89e678962122e6d3dd4c0046cdf Mon Sep 17 00:00:00 2001
From: lordpsyan <realmsofwarcraft@gmail.com>
Date: Sun, 30 Dec 2012 19:00:21 -0600
Subject: [PATCH] 2012_12_30-garments-of-the-light

---
 src/game/Creature.cpp |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/src/game/Creature.cpp b/src/game/Creature.cpp
index 58d2903..c9096e8 100644
--- a/src/game/Creature.cpp
+++ b/src/game/Creature.cpp
@@ -390,6 +390,8 @@ bool Creature::UpdateEntry(uint32 Entry, Team team, const CreatureData* data /*=
         else
             SetPvP(false);
     }
+    if (GetCreatureInfo()->unit_flags & UNIT_FLAG_PVP || GetCreatureInfo()->unit_flags & UNIT_FLAG_PVP_ATTACKABLE)
+    SetPvP(true);
 
     for (int i = 0; i < CREATURE_MAX_SPELLS; ++i)
         m_spells[i] = GetCreatureInfo()->spells[i];
-- 
1.7.2.5

