diff -r dc4fc5c8b306 src/server/game/Entities/Creature/Creature.cpp
--- a/src/server/game/Entities/Creature/Creature.cpp	Tue Nov 30 21:32:00 2010 +0100
+++ b/src/server/game/Entities/Creature/Creature.cpp	Tue Nov 30 22:11:00 2010 -0500
@@ -2217,10 +2217,24 @@
             return;
         }
     }
-
-    x = GetPositionX();
-    y = GetPositionY();
-    z = GetPositionZ();
+	if (m_homePosition.IsPositionValid())
+	{
+	    
+	    if((m_homePosition.GetPositionX() != 0) && (m_homePosition.GetPositionY() != 0) && (m_homePosition.GetPositionZ() != 0) )
+	    {
+	        x = m_homePosition.GetPositionX();
+            y = m_homePosition.GetPositionY();
+            z = m_homePosition.GetPositionZ();
+			if (ori)
+                *ori = GetOrientation();
+            if (dist)
+                *dist = 0;
+			return;
+		}
+	}
+	x = GetPositionX();
+	y = GetPositionY();
+	z = GetPositionZ();
     if (ori)
         *ori = GetOrientation();
     if (dist)
