diff -r dc4fc5c8b306 src/server/game/Globals/ObjectMgr.cpp
--- a/src/server/game/Globals/ObjectMgr.cpp	Tue Nov 30 21:32:00 2010 +0100
+++ b/src/server/game/Globals/ObjectMgr.cpp	Tue Nov 30 23:03:15 2010 -0500
@@ -2089,7 +2089,7 @@
             }
         }
         else
-            sLog.outErrorDb("Item (Entry: %u) does not exist in item.dbc! (not correct id?).",i);
+            // sLog.outErrorDb("Item (Entry: %u) does not exist in item.dbc! (not correct id?).",i);
 
         if (proto->Class >= MAX_ITEM_CLASS)
         {
diff -r dc4fc5c8b306 src/server/game/Grids/GridDefines.h
--- a/src/server/game/Grids/GridDefines.h	Tue Nov 30 21:32:00 2010 +0100
+++ b/src/server/game/Grids/GridDefines.h	Tue Nov 30 23:03:15 2010 -0500
@@ -31,7 +31,7 @@
 class Pet;
 class Player;
 
-#define MAX_NUMBER_OF_CELLS     8
+#define MAX_NUMBER_OF_CELLS     1
 
 #define MAX_NUMBER_OF_GRIDS      64
 
diff -r dc4fc5c8b306 src/server/game/World/World.cpp
--- a/src/server/game/World/World.cpp	Tue Nov 30 21:32:00 2010 +0100
+++ b/src/server/game/World/World.cpp	Tue Nov 30 23:03:15 2010 -0500
@@ -732,12 +732,6 @@
     }
     else
         m_int_configs[CONFIG_MAX_PLAYER_LEVEL] = sConfig.GetIntDefault("MaxPlayerLevel", DEFAULT_MAX_LEVEL);
-
-    if (m_int_configs[CONFIG_MAX_PLAYER_LEVEL] > MAX_LEVEL)
-    {
-        sLog.outError("MaxPlayerLevel (%i) must be in range 1..%u. Set to %u.",m_int_configs[CONFIG_MAX_PLAYER_LEVEL],MAX_LEVEL,MAX_LEVEL);
-        m_int_configs[CONFIG_MAX_PLAYER_LEVEL] = MAX_LEVEL;
-    }
 
     m_int_configs[CONFIG_MIN_DUALSPEC_LEVEL] = sConfig.GetIntDefault("MinDualSpecLevel", 40);
 
