diff -r 51b295fedcc7 src/server/scripts/EasternKingdoms/silvermoon_city.cpp
--- a/src/server/scripts/EasternKingdoms/silvermoon_city.cpp	Wed Dec 22 00:12:03 2010 +0100
+++ b/src/server/scripts/EasternKingdoms/silvermoon_city.cpp	Tue Dec 21 23:38:14 2010 -0500
@@ -28,7 +28,7 @@
 EndContentData */
 
 #include "ScriptPCH.h"
-
+#define NPC_BLOOD_KNIGHT_STILLBLADE    17768
 /*#######
 # npc_blood_knight_stillblade
 #######*/
@@ -90,7 +90,7 @@
             if ((Spellkind->Id == SPELL_SHIMMERING_VESSEL) && !spellHit &&
                 (Hitter->GetTypeId() == TYPEID_PLAYER) && (CAST_PLR(Hitter)->IsActiveQuest(QUEST_REDEEMING_THE_DEAD)))
             {
-                CAST_PLR(Hitter)->AreaExploredOrEventHappens(QUEST_REDEEMING_THE_DEAD);
+                CAST_PLR(Hitter)->KilledMonsterCredit(NPC_BLOOD_KNIGHT_STILLBLADE, 0);
                 DoCast(me, SPELL_REVIVE_SELF);
                 me->SetStandState(UNIT_STAND_STATE_STAND);
                 me->SetUInt32Value(UNIT_DYNAMIC_FLAGS, 0);
