diff --git a/src/game/Creature.cpp b/src/game/Creature.cpp
index 90cfa9c..56a2a49 100644
--- a/src/game/Creature.cpp
+++ b/src/game/Creature.cpp
@@ -389,6 +389,9 @@ bool Creature::UpdateEntry(uint32 Entry, Team team, const CreatureData *data /*=
             SetPvP(false);
     }
 
+ if (GetCreatureInfo()->unit_flags & UNIT_FLAG_PVP || GetCreatureInfo()->unit_flags & UNIT_FLAG_PVP_ATTACKABLE)
+ SetPvP(true);
+
     for(int i = 0; i < CREATURE_MAX_SPELLS; ++i)
         m_spells[i] = GetCreatureInfo()->spells[i];