+++ b/src/bindings/ScriptDev2/scripts/eastern_kingdoms/zulaman/zulaman.h
@@ -151,7 +151,7 @@ struct NalorakkBearEventInfo
 
 static const NalorakkBearEventInfo aBearEventInfo[MAX_BEAR_WAVES] =
 {
-    {SAY_WAVE1_AGGRO,    0, 0, 0, 0, 45.0f},
+    {SAY_WAVE1_AGGRO,    13.3220f, 1416.299f, 11.890f, 5.867f, 45.0f},
     {SAY_WAVE2_STAIR1,   -54.948f, 1419.772f, 27.303f, 0.03f, 37.0f},
     {SAY_WAVE3_STAIR2,   -80.303f, 1372.622f, 40.764f, 1.67f, 35.0f},
     {SAY_WAVE4_PLATFORM, -77.495f, 1294.760f, 48.487f, 1.66f, 60.0f}
	 